libMachO/MachO_ABI/LoadCommands/load_command_linker_option.c

```diff
@@ -0,0 +1,167 @@
+
+#include "macho_abi_internal.h"
+
+//|++++++++++++++++++++++++++++++++++++|//
+static size_t
+_mk_load_command_linker_option_copy_description(__unused mk_load_command_ref load_command, char *output, size_t output_len)
+{
+    char all_options_buffer[512];
+    bzero(all_options_buffer, sizeof(all_options_buffer));
+    
+    // invariant:
+    //   - chars at "strings_length" and beyond are all NULL
+    //   - chars at indexes less than "strings_length" are non-NULL
+    //   - strings_length < sizeof(description_buffer)
+    size_t strings_length = 0;
+    uint32_t num_strings = mk_load_command_linker_option_get_nstrings(load_command);
+    if (num_strings == UINT32_MAX) {
+        return 0;
+    }
+    
+    for (uint32_t i = 0; i < num_strings; ++i) {
+        char current_option_buffer[512];
+        size_t copy_result = mk_load_command_linker_option_copy_string(load_command, i, current_option_buffer, sizeof(current_option_buffer));
+        if (copy_result == 0) {
+            return 0;
+        }
+        
+        size_t remaining_length = sizeof(all_options_buffer) - strings_length;
+        size_t char_count = (size_t)snprintf(&all_options_buffer[strings_length], remaining_length, "\t%s\n", current_option_buffer);
+        if (char_count + 1 /* NULL char */ > remaining_length) {
+            return 0;
+        }
+        
+        strings_length += char_count; // point at the first NULL char
+    }
+    
+    return (size_t)snprintf(output, output_len, "<%s %p> {\n%s}",
+                                mk_type_name(load_command.type), load_command.type,
+                                all_options_buffer);
+}
+
+const struct _mk_load_command_vtable _mk_load_command_linker_option_class = {
+    .base.super                 = &_mk_load_command_class,
+    .base.name                  = "LC_LINKER_OPTION",
+    .base.copy_description      = &_mk_load_command_linker_option_copy_description,
+    .command_id                 = LC_LINKER_OPTION,
+    .command_base_size          = sizeof(struct linker_option_command)
+};
+
+//|++++++++++++++++++++++++++++++++++++|//
+uint32_t mk_load_command_linker_option_id()
+{ return LC_LINKER_OPTION; }
+
+uint32_t
+mk_load_command_linker_option_get_nstrings(mk_load_command_ref load_command)
+{
+    _MK_LOAD_COMMAND_NOT_NULL(load_command, return UINT32_MAX);
+    _MK_LOAD_COMMAND_IS_A(load_command, _mk_load_command_linker_option_class, return UINT32_MAX);
+    
+    struct linker_option_command *linker_option_command = (struct linker_option_command*)load_command.load_command->mach_load_command;
+    return mk_macho_get_byte_order(load_command.load_command->image)->swap32( linker_option_command->count );
+}
+
+static const char*
+mk_load_command_linker_option_get_string(mk_load_command_ref load_command, uint32_t requested_index, size_t *string_length) {
+    _MK_LOAD_COMMAND_NOT_NULL(load_command, return NULL);
+    _MK_LOAD_COMMAND_IS_A(load_command, _mk_load_command_linker_option_class, return NULL);
```

Comment:
`mk_load_command_linker_option_get_string` is a private function (not in the header).  The null+type checks only need to appear in the API entry points (`mk_load_command_linker_option_enumerate_strings`).